### PR TITLE
Update ffmpeg

### LINKED ISSUE #44: FFMPEG dependency update
**Describe the bug**

Hi guys I would like to update ffmpeg library, but I'm facing this issues:
- CBS is now included into ffmpeg.
- cbs.h, cbs_h264.h and cbs_h265.h aren't copied into local64/include/libavcodec directory.
- Sunshine compilation fails during linking as cbs.o, cbs_h264.o and cbs_h265.o has been statically linked with ffmpeg. It conflicts with Sunshine's cbs.

Replacing current CBS implementation by ffmpeg one is a solution, only if I can fix the first issue.

I don't know how to add parameters to [media-autobuild_suite](https://github.com/m-ab-s/media-autobuild_suite) so it can pass it to ffmpeg autoconf. It seems that options are into ffmpeg's CONFIG_EXTRA variable, but we can't set them externaly. Plus I don't know if it will copy right headers file to include directory.

**To Reproduce**
Only for development purpose, then we must use Sunshine's Workflows.
1. Execute [media-autobuild_suite](https://github.com/m-ab-s/media-autobuild_suite) batch file
2. Create a sunshine-prebuild.zip file (see [SunshineStream/Sunshine-Windows-Prebuilt](https://github.com/SunshineStream/Sunshine-Windows-Prebuilt) workflow to replicate the process localy)
3. Comment CMakeList.txt sunshine-prebuild.zip
4. Copy sunshine-prebuild.zip into Sunshine's build directory
4. Execute Sunshine build.

**Expected behavior**
Sunshine build and run correctly.

### PR DESCRIPTION
**Pull request by PapyKahan**
## Description

Changes to be able to update FFMPEG dependency. I've removed duplicated implementation of cbs. There's an implementation into libavcodec now. I've left existing headers to be able to build and expose cbs implementation of FFMPEG.

I've also fix a segmentation fault introduced with FFMPEG update by setting default values into AVPacket struct.

### Issues Fixed or Closed

- Fixes #44

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the documentation blocks for new or existing components

### CODE REVIEW

File: sunshine/video.h [hunk 1]
```diff
@@ -27,6 +27,8 @@ struct packet_raw_t : public AVPacket {
     buf             = nullptr;
     side_data       = nullptr;
     side_data_elems = 0;
+    opaque          = nullptr;
+    opaque_ref      = nullptr;
```
Comment (ReenigneArcher) [inline sunshine/video.h:31]: I believe, these two lines are causing the build to fail

**Comment by ReenigneArcher** [comment]
So, looks like Windows is building okay. What do we need to do to get the building on Linux functional again? Can we use a pre-built like on Windows?

**Comment by ReenigneArcher** [comment]
> So, looks like Windows is building okay. What do we need to do to get the building on Linux functional again? Can we use a pre-built like on Windows?

I made some cuda version updates in this branch: https://github.com/SunshineStream/Sunshine/tree/update-ffmpeg-linux

Not sure it will resolve building on linux, worth a shot to see if we get any further. Want to merge that branch into your `udpate-ffmpeg` branch?

I also have a question. Are these changes to ffmpeg only required for building, or also required during runtime?

**Comment by ReenigneArcher** [comment]
@PapyKahan I've updated the prebuild repo to include a build of ffmpeg on debian. Also the repo name changed.

Would you mind updating to point to the `-windows` package? And do you have any idea about building properly on linux? I'm hoping that `-debian` package can resolve the issues we're seeing in the build here.

**Comment by GeoDerp** [comment]
Im not sure if this will help but while testing my rpm spec #7 with this branch I get 

```shell
/home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:30:5: error: 'opaque' was not declared in this scope
   30 |     opaque          = nullptr;
      |     ^~~~~~
/home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:31:5: error: 'opaque_ref' was not declared in this scope
   31 |     opaque_ref      = nullptr;
      |     ^~~~~~~~~~
```

**Comment by ReenigneArcher** [comment]
> Im not sure if this will help but while testing my rpm spec #7 with this branch I get
> 
> ```shell
> /home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:30:5: error: 'opaque' was not declared in this scope
>    30 |     opaque          = nullptr;
>       |     ^~~~~~
> /home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:31:5: error: 'opaque_ref' was not declared in this scope
>    31 |     opaque_ref      = nullptr;
>       |     ^~~~~~~~~~
> ```

We're aware of the issue. https://github.com/SunshineStream/Sunshine/runs/5174320304?check_suite_focus=true

**Comment by GeoDerp** [comment]
> > Im not sure if this will help but while testing my rpm spec #7 with this branch I get
> > ```shell
> > /home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:30:5: error: 'opaque' was not declared in this scope
> >    30 |     opaque          = nullptr;
> >       |     ^~~~~~
> > /home/Geo/rpmbuild/BUILD/sunshine-master/sunshine/video.h:31:5: error: 'opaque_ref' was not declared in this scope
> >    31 |     opaque_ref      = nullptr;
> >       |     ^~~~~~~~~~
> > ```
> 
> We're aware of the issue. https://github.com/SunshineStream/Sunshine/runs/5174320304?check_suite_focus=true

sorry, that was me being dumb and not using ffmpeg-5

**Pull request closed without merging**